app.php

```diff
@@ -66,7 +66,11 @@
 if (!file_exists($hashCacheFile))
 {
 	EmptyFolder($viewcachePath);
-	touch($hashCacheFile);
+	if!(touch($hashCacheFile))
+	{
+		// We failed to create the cache file, probably because of a missing write permission, bail!
+		exit('Unable to write to the viewcache directory.');
```

Comment:
Maybe better as `exit("Unable to write to the data/viewcache directory.\n");` ... specify the directory a bit more and include a newline for people who are using command line / curl to debug.